### PR TITLE
GoMod: Distinguish main module from development-only dependencies

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

These are improvements on top of recent work done for fixing #5021.

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml [hunk 1]
```diff
@@ -16,6 +16,20 @@ project:
     path: "<REPLACE_PATH>"
   homepage_url: ""
   scopes:
+  - name: "dependencies"
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml:19]: Commit message: Typo in "developmenti-only".

File: analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ project:
       linkage: "PROJECT_STATIC"
     - id: "GoMod::golang.org/x/sys:v0.0.0-20220610221304-9f5ed59c137d"
       linkage: "PROJECT_STATIC"
-  - name: "vendor"
+  - name: "devDependencies"
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml:33]: I'm not sure whether this renaming is a good idea. Classically, "vendored" dependencies are dependencies that were copied to the actual project's source code. At least that's how I know the term from other languages, like Ruby. And isn't this also the case for Go? Or are these dependencies *not* copied to the project, and really retrieved from a remote, like "regular" dependencies?

Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml:33]: The "vendor" package can be copied to a "vendor" directory under the project root using the vendor command.
Looking at the following links, in my view it's ok to interpret it as the set of packages which actually do get copies. These are exactly the set of packages / modules needed to build and test the main module. Note that these are exactly the set of dependencies which should be included into ORT's analysis result.

Maybe "all" could be a better name to use for that "vendor" / "devDependencies" scope.
Anyhow, I suggest we just try to find good names for these two scopes. I'm open for changing these (have no strong opinion).

1. https://go.dev/ref/mod#vendoring
2. https://github.com/golang/go/blob/2a78e8afc0994f5b292bc9a5a7258c749e43032f/src/cmd/go/internal/modcmd/why.go#L26-L28

File: analyzer/src/main/kotlin/managers/GoMod.kt [hunk 3]
```diff
@@ -199,6 +211,26 @@ class GoMod(
...
+        )
+
+        list.stdout.lines().forEach { line ->
+            val columns = line.split(' ')
+            if (columns.size != 2) return@forEach
+
+            result += columns[0]
+        }
+
+        return result
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/GoMod.kt:231]: You can make it a bit more compact:
```kotlin
return list.stdout.lineSequence()
    .map { it.split(' ') }
    .filter { it.size == 2 }
    .map { it[0] }
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/GoMod.kt (hunk 3 above)
Comment (fviernau) [reply analyzer/src/main/kotlin/managers/GoMod.kt:231]: I believe as-is is more consistent with local style. So in this commit I prefer to keep it. Fine?

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**